File: 0238-product-of-array-except-self/0238-product-of-array-except-self.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int[] prefix = new int[nums.length];
        prefix[0] = 1;
        for(int i=1; i<nums.length; i++) {
            prefix[i] = prefix[i-1]*nums[i-1];
        }
        int postfix = 1;

        for(int i = nums.length-2; i>=0; i--) {
            postfix *= nums[i+1];
            prefix[i] *= postfix;
        }
        return prefix;
    }
}

File: 0238-product-of-array-except-self/0238-product-of-array-except-self.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        prefix = [1]*len(nums)
        for i in range(1, len(nums), 1):
            prefix[i] = nums[i-1]*prefix[i-1]
        postfix = 1

        for i in range(len(nums)-2, -1, -1):
            prefix[i+1] *= postfix
            postfix *= nums[i+1]
        prefix[0] *= postfix
        return prefix